### ISSUE TITLE
thread 'rustc' panicked at 'Trying to convert unsized value to lval'

### ISSUE THREAD

**Issue by SimonSapin** [rust-lang/rust#22659]
[rust-casefold](https://github.com/SimonSapin/rust-casefold/tree/7fffc29797eb8bffe8f8bf4633212cfc63f27e56) used to pass its tests every night on Travis-CI, until a few nightlies ago. In rustc 1.0.0-nightly (522d09dfe 2015-02-19) (built 2015-02-21), it now causes an ICE:

```
% RUST_BACKTRACE=1 cargo build --verbose
       Fresh regex v0.1.15
   Compiling caseless v0.0.1 (file:///home/simon/projects/rust-casefold)
     Running `rustc src/lib.rs --crate-name caseless --crate-type lib -g -C metadata=8dd51172a0ed213f -C extra-filename=-8dd51172a0ed213f --out-dir /home/simon/projects/rust-casefold/target --emit=dep-info,link -L dependency=/home/simon/projects/rust-casefold/target -L dependency=/home/simon/projects/rust-casefold/target/deps`
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Trying to convert unsized value to lval', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/librustc_trans/trans/datum.rs:495

stack backtrace:
   1:     0x7f837e8b3210 - sys::backtrace::write::h252031bd050bf19aKlC
   2:     0x7f837e8dbac0 - panicking::on_panic::h8a07e978260e2c7btXL
   3:     0x7f837e81c720 - rt::unwind::begin_unwind_inner::h322bcb3f35268c19RBL
   4:     0x7f837d4c8590 - rt::unwind::begin_unwind::h8757258989833155500
   5:     0x7f837d5411e0 - trans::datum::Datum<'tcx, Expr>::to_lvalue_datum::h4c859596f8252b6ehYf
   6:     0x7f837d5737c0 - trans::expr::trans_index::h700966d3e1a7a6016Ei
   7:     0x7f837d560dd0 - trans::expr::trans_unadjusted::h964cf47356d964c2sqi
   8:     0x7f837d518360 - trans::expr::trans::ha0e2a08ab1f695a2HMh
   9:     0x7f837d578be0 - trans::expr::trans_field::h10880054022304515065
  10:     0x7f837d560dd0 - trans::expr::trans_unadjusted::h964cf47356d964c2sqi
  11:     0x7f837d518360 - trans::expr::trans::ha0e2a08ab1f695a2HMh
  12:     0x7f837d516cc0 - trans::expr::trans_into::hbe2ad59d17caa991pGh
  13:     0x7f837d595e60 - trans::expr::trans_adt::h5375342382b003c3zvj
  14:     0x7f837d551600 - trans::base::trans_named_tuple_constructor::h67d0c2e9e4926af7vwu
  15:     0x7f837d558040 - trans::callee::trans_call_inner::h14909791323518108589
  16:     0x7f837d561990 - trans::expr::trans_rvalue_dps_unadjusted::h2f62e3fa23f391ad9Wi
  17:     0x7f837d516cc0 - trans::expr::trans_into::hbe2ad59d17caa991pGh
  18:     0x7f837d603260 - trans::_match::trans_match_inner::h755825b289633425aMx
  19:     0x7f837d561990 - trans::expr::trans_rvalue_dps_unadjusted::h2f62e3fa23f391ad9Wi
  20:     0x7f837d516cc0 - trans::expr::trans_into::hbe2ad59d17caa991pGh
  21:     0x7f837d517830 - trans::controlflow::trans_block::h717e6f4441ba9a815ee
  22:     0x7f837d5e4d70 - trans::base::trans_closure::h966338b3741dc0e7phu
  23:     0x7f837d503b30 - trans::base::trans_fn::hd5e20febd5f53650Gsu
  24:     0x7f837d4fed50 - trans::base::trans_item::h5bf53079c2f2baddzRu
  25:     0x7f837d5ec2e0 - trans::base::trans_crate::h22d6b6c0d890fedf1Ov
  26:     0x7f837ef3a290 - driver::phase_4_translate_to_llvm::h9c18e73369736a67qPa
  27:     0x7f837ef14070 - driver::compile_input::h0c8d8120f6194473Gba
  28:     0x7f837efe32b0 - run_compiler::heccd2f43b844857cZbc
  29:     0x7f837efe1bb0 - thunk::F.Invoke<A, R>::invoke::h14556331494175157254
  30:     0x7f837efe0aa0 - rt::unwind::try::try_fn::h8181016918202193540
  31:     0x7f837e948880 - rust_try_inner
  32:     0x7f837e948870 - rust_try
  33:     0x7f837efe0da0 - thunk::F.Invoke<A, R>::invoke::h12418455792421810631
  34:     0x7f837e8c7b60 - sys::thread::thread_start::h3defdaea150d8cd693G
  35:     0x7f83787c92b0 - start_thread
  36:     0x7f837e4a5249 - __clone
  37:                0x0 - <unknown>

Could not compile `caseless`.

Caused by:
  Process didn't exit successfully: `rustc src/lib.rs --crate-name caseless --crate-type lib -g -C metadata=8dd51172a0ed213f -C extra-filename=-8dd51172a0ed213f --out-dir /home/simon/projects/rust-casefold/target --emit=dep-info,link -L dependency=/home/simon/projects/rust-casefold/target -L dependency=/home/simon/projects/rust-casefold/target/deps` (status=101)
```

**Comment by SimonSapin** [comment]
Possibly duplicate of #22659.

**Comment by SimonSapin** [comment]
This seems fixed in rustc 1.0.0-nightly (4db0b3246 2015-02-25) (built 2015-02-26).
